common4j/src/main/com/microsoft/identity/common/java/crypto/StorageEncryptionManager.java

```diff
@@ -0,0 +1,530 @@
+            exception = e;
+        } catch (InvalidAlgorithmParameterException e) {
+            errCode = INVALID_ALG_PARAMETER;
+            exception = e;
+        }
+
+        throw new ClientException(errCode, exception.getMessage(), exception);
+    }
+
+    @Override
+    @NonNull
+    public byte[] decrypt(@NonNull final byte[] cipherText) throws ClientException {
+        final String methodName = ":decrypt";
+        Logger.verbose(TAG + methodName, "Starting decryption");
+
+        final List<AES256KeyLoader> keysForDecryption = getKeyLoaderForDecryption(cipherText);
+
+        if (keysForDecryption == null || keysForDecryption.size() == 0) {
+            if (keysForDecryption == null){
+                throw new IllegalStateException("KeyLoader list must not be null.");
+            }
+        }
+
+        final byte[] dataBytes;
+        try {
+            dataBytes = stripEncodeVersionFromCipherText(cipherText);
+        } catch (final ClientException e){
+            Logger.verbose(TAG + methodName,
+                    "Failed to strip encode version from cipherText, string might not be encrypted. Exception: ", e.getMessage());
+            return cipherText;
+        }
+
+        final ClientException exceptionToThrowIfAllFails = new ClientException(ErrorStrings.DECRYPTION_FAILED,
+                "Tried all decryption keys and decryption still fails.");
+
+        for (final ISecretKeyLoader keyLoader : keysForDecryption) {
+            if (keyLoader == null){
+                throw new IllegalStateException("KeyLoader must not be null.");
+            }
+            
+            try {
+                final byte[] result = decryptWithSecretKey(dataBytes, keyLoader);
+                Logger.verbose(TAG + methodName, "Finished decryption with key:" + keyLoader.getAlias());
+                return result;
+            } catch (ClientException e) {
+                Logger.warn(TAG + methodName, "Failed to decrypt with key:" + keyLoader.getAlias() +
+                        " thumbprint : " + KeyUtil.getKeyThumbPrint(keyLoader));
+                handleDecryptionFailure(keyLoader.getAlias(), e);
+                exceptionToThrowIfAllFails.addSuppressed(e);
+            }
+        }
+
+        Logger.warn(
+                TAG + methodName,
+                exceptionToThrowIfAllFails.getMessage());
+
+        throw exceptionToThrowIfAllFails;
+    }
+
+    private void tryLoggingKeyThumbPrint(@NonNull final ISecretKeyLoader keyLoader){
+
+    }
+
+    @Override
+    public byte[] sign(byte[] text) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean verify(byte[] text, byte[] signature) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public byte[] getThumbprint() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Certificate[] getCertificateChain() {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
Some details about why these are unsupported would be nice

Reply:
Seems this is marked as resolved...but I'm not seeing the requested details...not sure where these were added

Reply:
Can we just not override these methods as the class is `abstract` anyway?